soccer/src/vision/VisionFilter.hpp

```diff
@@ -51,16 +46,33 @@ class VisionFilter {
      */
     void fillRobotState(SystemState& state, bool usBlue);
 
+    /**
+     * @brief Returns the latest timestamp of the received vision packets.
+     * @return
+     */
+    [[nodiscard]] RJ::Time GetLastVisionTime() const {
+        return RJ::FromROSTime(last_update_time_);
+    }
+
 private:
     void updateLoop();
 
+    /**
+     * @brief Obtains a std::vector<DetectionFrameMsg::UniqePtr> from
+     * detection_frame_sub_, then converts that to std::vector<CameraFrame>
+     * and stores it in new_frames_.
+     */
+    void GetFrames();
+
     std::thread worker;
 
     std::mutex worldLock;
     World world;
 
     std::atomic_bool threadEnd{};
 
-    std::mutex frameLock;
     std::vector<CameraFrame> frameBuffer{};
+    Context* context_;
+    std::unique_ptr<ros2_temp::DetectionFrameSub> detection_frame_sub_;
+    rclcpp::Time last_update_time_;
```

Comment:
@kylestach thoughts on `RJ::Time` vs `rclcpp::Time`?

Reply:
So, I have mixed thoughts on this.

I _really_ think we should be careful about how we use `Time` types. You should pretty much _never_ call `now()` (or equivalent) for anything meaningful except at very defined boundaries (i.e. at the node level) - this makes it a lot easier to mock things out later on. I think ROS does a really good job of enforcing this by forcing you to have a `Node` in order to actually get the current time, but on the other hand limiting ROS types except at the boundaries is a Good Thing™.

I think a good compromise-ish would be to never call `RJ::now()` (use the ROS equivalent instead) but to use `RJ::Time` internally.

Tangentially, `RJ::Seconds` is evil; change my mind.

Reply:
I'm curious now, what's the issue with `RJ::Seconds`?

Reply:
I mean in general it's not like it shouldn't be used at all, we just use it far too often to represent "some time in the future" when we really should be using an `RJ::Time`. I think at least half of it's usage in our codebase (at least pre-motion planning rewrite) should have been replaced with a `Time`.

Reply:
Any examples of this misuse?

Reply:
https://github.com/RoboJackets/robocup-software/blob/8ed909cf54317f176b4ba71369532897228af4c9/soccer/planning/trajectory/Trajectory.hpp#L65

Reply:
I think we should create our own `RJ::Time` and `RJ::Duration` that isn't just a type alias. I think if we call it `RJ::Duration` then it'll be easier to not misuse, ie. `RJ::Duration initialTime` just looks and sounds so much more wrong than `RJ::Seconds initialTime` (Who knows, it might be seconds since epoch or something).